File: src/js/tests/noteViewPhrase.test.ts
import { expect, test } from 'vitest';
import { NoteViewPhrase, Pitch, Raga } from '@model';

test('NoteViewPhrase basic', () => {
  const r = new Raga();
  const nv = new NoteViewPhrase({ pitches: [new Pitch()], durTot: 1, raga: r, startTime: 0 });
  expect(nv.pitches.length).toBe(1);
  expect(nv.durTot).toBe(1);
  expect(nv.raga).toBeInstanceOf(Raga);
  expect(nv.startTime).toBe(0);
});

File: python/api/tests/note_view_phrase_test.py
from python.api.classes.note_view_phrase import NoteViewPhrase
from python.api.classes.pitch import Pitch
from python.api.classes.raga import Raga


def test_note_view_phrase_basic():
    r = Raga()
    nv = NoteViewPhrase({'pitches': [Pitch()], 'dur_tot': 1, 'raga': r, 'start_time': 0})
    assert len(nv.pitches) == 1
    assert nv.dur_tot == 1
    assert isinstance(nv.raga, Raga)
    assert nv.start_time == 0
